internal/internal_worker_base.go

```diff
@@ -401,3 +410,38 @@ func (bw *baseWorker) Stop() {
 	}
 	return
 }
+
+func (bw *baseWorker) EmitHardwareUsage() {
+	defer bw.shutdownWG.Done()
```

Comment:
I think this needs to be in the goroutine

Reply:
wrapped entire function in a goroutine now